eventmesh-common/src/main/java/org/apache/eventmesh/common/protocol/tcp/codec/Codec.java

```diff
@@ -161,7 +159,7 @@ private Header parseHeader(ByteBuf in, int headerLength) throws JsonProcessingEx
             }
             final byte[] headerData = new byte[headerLength];
             in.readBytes(headerData);
-            LogUtils.debug(log, "Decode headerJson={}", deserializeBytes(headerData));
+            log.atDebug().log("Decode headerJson={}", deserializeBytes(headerData));
```

Comment:
```suggestion
            log.atDebug().addArgument(() -> deserializeBytes(headerData)).log("Decode headerJson={}");
```

Since the purpose of your PR is to prevent expensive computations of disabled log statement arguments, you should use a lambda here.

Reply:
True🤣I tested it with the code below, and the timer shows that only Supplier could avoid executing `formatMsg()`. However the Fluent API said that `For disabled log levels, the returned LoggingEventBuilder instance does nothing`. Why is that?

Test code, ran under `debug` level:

```
        ......
        long traceStart = System.currentTimeMillis();
        log.trace("log.trace(). formatMsg() return value: {}", formatMsg("log.trace()", "executed"));
        long traceEnd = System.currentTimeMillis();
        long atTraceStart = System.currentTimeMillis();
        log.atTrace().log("log.atTrace().log(). formatMsg() return value: {}", formatMsg("log.atTrace().log()", "executed"));
        long atTraceEnd = System.currentTimeMillis();
        long atTraceSupplierStart = System.currentTimeMillis();
        log.atTrace().addArgument(() -> formatMsg("log.atTrace() Supplier", "executed")).log("log.atTrace().addArgument().log()");
        long atTraceSupplierEnd = System.currentTimeMillis();
        log.info("log.trace() cost: {}ms, log.atTrace().log() cost: {}ms, log.atTrace() Supplier cost: {}ms",
                traceEnd - traceStart, atTraceEnd - atTraceStart, atTraceSupplierEnd - atTraceSupplierStart);
    }

    private String formatMsg(String name, String status) {
        try {
            Thread.sleep(1000);
        } catch (InterruptedException e) {
            throw new RuntimeException(e);
        }
        return name + "-" + status;
    }
```

Output:

```
log.trace() cost: 1009ms, log.atTrace().log() cost: 1002ms, log.atTrace() Supplier cost: 0ms
```

Reply:
> True🤣I tested it with the code below, and the timer shows that only Supplier could avoid executing `formatMsg()`. However the Fluent API said that `For disabled log levels, the returned LoggingEventBuilder instance does nothing`. Why is that?

This is due to the order in which Java evaluates methods: it always evaluates the arguments of a method before the method itself. In this concrete example:

```java
log.atDebug().log("Decode headerJson={}", deserializeBytes(headerData));
```

Java will:

1. Call `Logger#atDebug()` and obtain a `NOPLoggingEventBuilder`,
2. Call `deserializeBytes` and obtain a string,
3. Call `NOPLoggingEventBuilder#log` which does nothing.

The [Scala language](https://docs.scala-lang.org/tour/by-name-parameters.html) has call-by-name arguments that are evaluated only if they are used. In Java everything is call-by-value and the arguments are computed even if they are not used.